kv/etcd/etcd.go

```diff
@@ -269,6 +301,80 @@ func (c *client) CheckAndSet(key string, version int, v proto.Message) (int, err
 	return version + 1, nil
 }
 
+func (c *client) getCache(key string) (kv.Value, bool) {
+	c.cache.RLock()
+	v, ok := c.cache.Values[key]
+	c.cache.RUnlock()
+
+	return v, ok
+}
+
+func (c *client) mergeCache(key string, v *value) {
+	c.cache.Lock()
+	defer c.cache.Unlock()
+
+	cur, ok := c.cache.Values[key]
+	if !ok || v.isNewer(cur) {
+		c.cache.Values[key] = v
+		go c.syncFS()
```

Comment:
This may cause a race w.r.t. writing out the cache file if we have two successive updates coming in since we are syncing to filesystem asynchronously, no? Probably better to have a long-lived goroutine watch for cache updates via a channel or something alike